### PR TITLE
feat: Add Image/Multimodal Search to Gen App Builder Search Demo

### PR DESCRIPTION
**Pull request by holtskinner**

### CODE REVIEW

File: gen-app-builder/search-web-app/genappbuilder_utils.py [hunk 1]
```diff
@@ -14,9 +14,9 @@
 
 """Generative AI App Builder Utilities"""
 from os.path import basename
-from typing import List, Optional, Tuple
+from typing import Dict, List, Optional, Tuple
 
-from google.cloud import discoveryengine, discoveryengine_v1beta
+from google.cloud import discoveryengine_v1beta as discoveryengine
```
Comment (koverholt) [inline gen-app-builder/search-web-app/genappbuilder_utils.py:19]: 😎

File: gen-app-builder/search-web-app/templates/index.html [hunk 2]
```diff
@@ -74,6 +74,7 @@ <h1>Enterprise Search Demo - Widgets</h1>
         triggerId="{{s.name|lower|replace(' ', '-')}}"
       >
       </gen-search-widget>
+      <div class="search-engine-name"><b>{{s.name}}</b></div>
```
Comment (koverholt) [inline gen-app-builder/search-web-app/templates/index.html:77]: The UX here is a bit jarring when clicking on one of the text search boxes out of the many, even for me who is familiar with how that ES widget works as a full-screen popover.

Perhaps add a note at the top of the page that clicking on one of the text boxes will open up a full-page search widget for that particular engine? Or maybe a different UX element would help here such as tabs for each search engine?

**Review by koverholt — APPROVED** [review]
LGTM, updates all look good. Left a comment about the multiple search widgets page, but since that was not introduced in this PR, feel free to ignore or address in a followup PR.

File: gen-app-builder/search-web-app/templates/index.html (hunk 2 above)
Comment (holtskinner) [reply gen-app-builder/search-web-app/templates/index.html:77]: I would agree that a different design would make more sense. TBH, this whole app design could probably be redone to make it more "reactive", but I just tried to get it out quickly :P

I'll add a section at the top clarifying the behavior

**Comment by holtskinner** [comment]
Waiting until https://github.com/googleapis/google-cloud-python/pull/11528 is merged because this feature requires the updated client libraries to be pushed out.

**Pull request merged**